pkg/ebpf/dns_cache.go

```diff
@@ -200,17 +200,23 @@ func (v *dnsCacheVal) copy() []string {
 }
 
 type translation struct {
-	name string
-	ips  map[util.Address]struct{}
+	dns string
+	ips []util.Address
 }
 
 func newTranslation(domain []byte) *translation {
 	return &translation{
-		name: string(domain),
-		ips:  make(map[util.Address]struct{}),
+		dns: string(domain),
+		ips: make([]util.Address, 0),
 	}
 }
 
 func (t *translation) add(addr util.Address) {
-	t.ips[addr] = struct{}{}
```

Comment:
Since this tends to be small, it turns out that it's way cheaper (mainly in terms of memory allocations) to use a slice and do a brute-force search. Contrarily to a map we can re-use the same slice;